### ISSUE TITLE
Local asset filenames can be invalid for Android and Windows devices

### ISSUE THREAD

**Issue by DawMatt** [b-yp/logseq-link-to-local#20]
**Issue:** 
Please ensure the local asset filename used is valid on Windows and Android devices. Please avoid using characters such as asterisk (*), question mark (?), or slashes (/ or \) as part of the filename, as they are not valid filename characters on some operating systems.

This issue renders the plugin virtually unusable if you are using it on a Mac, with a Git based workflow, and expect the Logseq data to also be used on other platforms (e.g. Windows or Android).

**Context:**
I tried to reset my Git repo clone on my Android device to align with the main branch. It failed, with part of the log output included below:

```
error: unable to create file assets/storages/logseq-link-to-local/1*zRlkbxrRXjeTYeeail-FSA_49759_1_1702693950847.png: Operation not permitted
fatal: Could not reset index file to revision 'origin/main'.
```

Note the asset filename: `1*zRlkbxrRXjeTYeeail-FSA_49759_1_1702693950847.png`

The asterisk (*) character is not a valid filename character on Windows or Android. 

**Workaround:**
I've seen the same issue before when I tried to reset my Windows clone of the Git repo. The only way I could resolve this was use my Mac to manually rename the local assets in Git, manually rename Logseq references to these files to align with the new filenames, commit, and push the change.

**Comment by b-yp** [comment]
The asset name here corresponds to the asset name in the source link.

Appreciate the feedback; I will optimize it accordingly.  💞

**Comment by DawMatt** [comment]
Thanks @b-yp ! I had wondered whether you were generating, or reusing, the filenames. I'm a little surprised that websites are using such impractical filenames.

**Comment by b-yp** [comment]
Currently planning to stick with the original link's name but replace special characters with "-".

Do you have any brilliant ideas?

**Comment by DawMatt** [comment]
> Currently planning to stick with the original link's name but replace special characters with "-".

That sounds practical. Using a dash or underscore character to replace the invalid file name character should work. Upon further thought there are more characters we may need to be careful of include pipe (|), greater/less than (<>), and dollar ($). This list seems like a reasonable starting point to filter out/replace: https://en.wikipedia.org/wiki/Filename#In_Windows

**Comment by b-yp** [comment]
I noticed that you might have missed some specific checks, for instance, XXX. Therefore, I'm hesitant to merge hastily, primarily due to the substantial size of this pull request. Is it possible to break it down into smaller parts?

**Comment by b-yp** [comment]
Thank you very much for providing this list. I will commence now.

**Comment by DawMatt** [comment]
@b-yp , thanks. I can see you have released a new version of the plugin. Is it possible to get this updated in the plugin marketplace as well please? At this point the new version isn't being detected automatically by logseq.

**Comment by DawMatt** [comment]
I did install the plugin manually and the core functionality is still working OK. The rename asset functionality is working well too.

I tested using this url:
https://medium.com/mlearning-ai/the-chatgpt-list-of-lists-a-collection-of-1500-useful-mind-blowing-and-strange-use-cases-8b14c35eb

You'll notice that the images tend to include asterisk characters in the names. The make assets local functionality now turns those into underscores. Thanks for making this change so promptly!

**Comment by DawMatt** [comment]
Thanks @b-yp ! I can see this has been updated in the marketplace, I've just updated using standard Logseq functionality, and the plugin is working well. Thanks for the fast turn around on this request. Great work! I'm closing this issue as it has been resolved by v1.6.4 .
